qiskit_nature/problems/sampling/protein_folding/bead_contacts/contact_map_builder.py

```diff
@@ -0,0 +1,127 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Builds a contact map that stores contacts between beads in a peptide."""
+import collections
+import logging
+from typing import Tuple, DefaultDict, Dict
+
+from qiskit.opflow import PauliSumOp, OperatorBase
+
+from ..peptide.pauli_ops_builder import (
+    _build_pauli_z_op,
+    _build_full_identity,
+)
+from ..peptide.peptide import Peptide
+
+logger = logging.getLogger(__name__)
+
+
+def _create_contact_qubits(
+    peptide: Peptide,
+) -> Tuple[
+    DefaultDict[int, dict],
+    DefaultDict[int, dict],
+    DefaultDict[int, dict],
+    DefaultDict[int, dict],
+    int,
+]:
+    """
+    Creates Pauli operators for 1st nearest neighbor interactions
+
+    Args:
+        peptide: A Peptide object that includes all information about a protein.
+
+    Returns:
+        lower_main_upper_main, lower_side_upper_main, lower_main_upper_side, \
+           lower_side_upper_side, num_contacts: Tuple consisting of dictionaries
+                                    of Pauli operators for contacts/
+                                    interactions between a lower/upper bead from the main/side
+                                    chain and a lower/upper bead from the main/side chain and the
+                                    total number of contacts.
+    """
+    main_chain_len = len(peptide.get_main_chain)
+    side_chain = peptide.get_side_chain_hot_vector()
+
+    lower_main_upper_main: DefaultDict[int, Dict[int, OperatorBase]] = collections.defaultdict(dict)
+    lower_side_upper_main: DefaultDict[int, Dict[int, OperatorBase]] = collections.defaultdict(dict)
+    lower_main_upper_side: DefaultDict[int, Dict[int, OperatorBase]] = collections.defaultdict(dict)
+    lower_side_upper_side: DefaultDict[int, Dict[int, OperatorBase]] = collections.defaultdict(dict)
+
+    num_contacts = 0
+    num_qubits = 2 * (main_chain_len - 1)
+    full_id = _build_full_identity(num_qubits)
+    for lower_bead_id in range(1, main_chain_len - 3):  # first qubit is number 1
+        for upper_bead_id in range(lower_bead_id + 3, main_chain_len + 1):
```

Comment:
Where do these number offsets come from? Also, further down you use more constants (e.g. `>=  5`).
Should these be configurable? If not, maybe give them a variable name or at least add a comment, please

Reply:
I think they correspond to how the constraints and interactions in the problem are modeled and are rather fixed. @pbark ? But I agree it would be nice to come up with well-named constants. Might be challenging to find concise names though.

Reply:
If these are constants which should not be changed I am also fine with a comment explaining what each number means and why it takes a particular value. The method using these constants could itself link to the relevant paper (or section thereof) in the docstring.